devito/propagator.py

```diff
@@ -85,12 +85,7 @@ def __init__(self, name, nt, shape, stencils, factorized=None, spc_border=0, tim
 
         # Settings for performance profiling
         self.profile = profile
-
-        if self.profile:
-            self.add_local_var("time", "double")
-            self.pre_loop.append(cgen.Statement("struct timeval start, end"))
-            self.pre_loop.append(cgen.Assign("time", 0))
-            self.post_loop.append(cgen.PrintStatement("time: %f\n", "time"))
+        self.timings = np.zeros(4)
```

Comment:
I'm not convinced passing an array here is the best thing to do, since it requires a property to the Operator for each timing we might add later, and it also makes the code slightly confusing. Instead, could we not use a `ctypes.struct`, to which we can just add fields and that we return to the user as a single object? This would also make it easier to inspect in IPython, I think.